app/assets/javascripts/admin/addon/models/admin-plugin.js

```diff
@@ -57,16 +57,25 @@ export default class AdminPlugin {
     // translation, and we can handle things like SAML instead of showing them
     // as Saml from discourse-saml. We can fall back to the programattic version
     // though if needed.
+    let name;
     if (this.translatedCategoryName) {
-      return this.translatedCategoryName;
+      name = this.translatedCategoryName;
+    } else {
+      name = this.name
+        .split("-")
+        .map((word) => {
+          return capitalize(word);
```

Comment:
It feels like if we are heading down the road of making this human readable we should translate this string

Maybe just, if you can translate ... translate otherwise just fall back to this implementation? 

maybe a convention that discourse.plugins_name.name ... for this?

Reply:
That's what we are doing already with `translatedCategoryName`  :) You shouldn't see this version if the plugin metadata + setting category translation has been set up correctly.